src/jep/InvocationHandler.java

```diff
@@ -119,6 +122,14 @@ public Object invoke(Object proxy, Method method, Object[] args)
             throws Throwable {
         this.jep.isValidThread();
 
+        if (this.functionalInterface && (method.getModifiers() & Modifier.ABSTRACT) == 0) {
+            /*
+             * If it's a non-abstract (default) interface method,
+             * fallback to invoking the default implementation if we're a functional interface.
+             */
+            return method.invoke(proxy, args);
```

Comment:
This will call itself, I wrote a small test class to verify and I got a StackOverflowError

```java
public class Test{

    public static void main(String[] args) throws Throwable{
        try(Jep jep = new Jep()){
            jep.set("test", new Test());
            jep.eval("test.checkConsumer(lambda: True)");
        }
    }

    public void checkConsumer(Consumer<?> consumer){
        consumer.andThen(null);
    }

}
```